tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1103,7 +1123,20 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
 
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.insert(operands.begin() + 2, MaybeConstantFoldBias(bias));
+    HloInstruction *maybe_constant_folded_bias = MaybeConstantFoldBias(bias);
+    if (slice && bitcast) {
+      maybe_constant_folded_bias =
+          instr->AddInstruction(HloInstruction::CreateBitcast(
+              slice->shape(), maybe_constant_folded_bias));
+    }
+
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget ||
+        gemm->custom_call_target() == kCublasLtMatmulCallTarget) {
```

Comment:
Shouldn't we pad for cublas non-LT gemms as well?

Reply:
Resolved.